src/function/scalar/list/contains_or_position.cpp

```diff
@@ -117,71 +70,94 @@ static void NestedListContainsFunction(DataChunk &args, ExpressionState &state,
 	case PhysicalType::INT8:
-		TemplatedListContainsFunction<int8_t>(args, state, result);
+		TemplatedContainsOrPosition<int8_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::INT16:
-		TemplatedListContainsFunction<int16_t>(args, state, result);
+		TemplatedContainsOrPosition<int16_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::INT32:
-		TemplatedListContainsFunction<int32_t>(args, state, result);
+		TemplatedContainsOrPosition<int32_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::INT64:
-		TemplatedListContainsFunction<int64_t>(args, state, result);
+		TemplatedContainsOrPosition<int64_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::INT128:
-		TemplatedListContainsFunction<hugeint_t>(args, state, result);
+		TemplatedContainsOrPosition<hugeint_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::UINT8:
-		TemplatedListContainsFunction<uint8_t>(args, state, result);
+		TemplatedContainsOrPosition<uint8_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::UINT16:
-		TemplatedListContainsFunction<uint16_t>(args, state, result);
+		TemplatedContainsOrPosition<uint16_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::UINT32:
-		TemplatedListContainsFunction<uint32_t>(args, state, result);
+		TemplatedContainsOrPosition<uint32_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::UINT64:
-		TemplatedListContainsFunction<uint64_t>(args, state, result);
+		TemplatedContainsOrPosition<uint64_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::FLOAT:
-		TemplatedListContainsFunction<float>(args, state, result);
+		TemplatedContainsOrPosition<float, T, OP>(args, state, result);
 		break;
 	case PhysicalType::DOUBLE:
-		TemplatedListContainsFunction<double>(args, state, result);
+		TemplatedContainsOrPosition<double, T, OP>(args, state, result);
 		break;
 	case PhysicalType::VARCHAR:
-		TemplatedListContainsFunction<string_t>(args, state, result);
+		TemplatedContainsOrPosition<string_t, T, OP>(args, state, result);
 		break;
 	case PhysicalType::MAP:
 	case PhysicalType::STRUCT:
 	case PhysicalType::LIST:
-		NestedListContainsFunction(args, state, result);
+		TemplatedContainsOrPosition<int8_t, T, OP, true>(args, state, result);
 		break;
 	default:
 		throw NotImplementedException("This function has not been implemented for this type");
 	}
 }
 
-static unique_ptr<FunctionData> ListContainsBind(ClientContext &context, ScalarFunction &bound_function,
-                                                 vector<unique_ptr<Expression>> &arguments) {
+static void ListContainsFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	return ListContainsOrPosition<bool, ContainsFunctor>(args, state, result);
+}
+
+static void ListPositionFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	return ListContainsOrPosition<int32_t, PositionFunctor>(args, state, result);
+}
+
+template <bool IS_CONTAINS = true>
```

Comment:
You could also template this `LogicalType RETURN_TYPE` instead of `bool IS_CONTAINS`, and then set `bound_function.return_type  = RETURN_TYPE`